recipes/extra-cmake-modules/all/conanfile.py

```diff
@@ -8,49 +12,65 @@ class ExtracmakemodulesConan(ConanFile):
+        else:
+            self.tool_requires("cmake/[>=3.16]")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("extra-cmake-modules-{}".format(self.version), self._source_subfolder)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_HTML_DOCS"] = False
+        tc.cache_variables["BUILD_QTHELP_DOCS"] = False
+        tc.cache_variables["BUILD_MAN_DOCS"] = False
+        tc.cache_variables["BUILD_TESTING"] = False
 
-        # KB-H016: do not install Find*.cmake
-        tools.replace_path_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-            "install(FILES ${installFindModuleFiles} DESTINATION ${FIND_MODULES_INSTALL_DIR})", "")
+        if self.package_folder is None:
+            share_folder = "res"
+        else:
+            share_folder = os.path.join(self.package_folder, "res")
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_HTML_DOCS"] = False
-        self._cmake.definitions["BUILD_QTHELP_DOCS"] = False
-        self._cmake.definitions["BUILD_MAN_DOCS"] = False
-        self._cmake.definitions["SHARE_INSTALL_DIR"] = os.path.join(self.package_folder, "res")
-        self._cmake.configure(source_folder=os.path.join(self.source_folder, self._source_subfolder))
-        return self._cmake
+        tc.cache_variables["SHARE_INSTALL_DIR"] = share_folder
+        tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
-        cmake.build()
+        cm_folder = "{}-{}".format(self.name, self.version)
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=cm_folder)
 
     def package(self):
-        cmake = self._configure_cmake()
-        cmake.install()
-        self.copy("testhelper.h", src=os.path.join(self.source_folder, self._source_subfolder, "tests/ECMAddTests"), dst="res/tests")
-        self.copy("*", src=os.path.join(self.source_folder, self._source_subfolder, "LICENSES"), dst="licenses")
+        cm_folder = "{}-{}".format(self.name, self.version)
+        lic_folder_st = os.path.join(cm_folder, "LICENSES")
+        lic_folder = os.path.join(self.source_folder, lic_folder_st)
+        self.output.info("license folder: {}".format(lic_folder))
 
-    def package_info(self):
-        self.cpp_info.resdirs = ["res"]
-        self.cpp_info.builddirs = ["res/ECM/cmake", "res/ECM/kde-modules", "res/ECM/modules", "res/ECM/test-modules", "res/ECM/toolchain"]
+        lic_folder_inst = os.path.join(self.package_folder, "licenses")
+        copy(self, "*", src=lic_folder, dst=lic_folder_inst)
+
+        cmake = CMake(self)
+        cmake.install()
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def package_info(self):
+        # this package is CMake files, it doesn't need an extra one generating
```

Comment:
```suggestion
        self.cpp_info.bindirs = []
        self.cpp_info.libdirs = []
```

Both library and executable folders are not used here, right?